lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
+        self.channel = channel
+
+        if channel:
+            self.select_callback.default_values = [channel]
+
+    @discord.ui.select(cls=discord.ui.ChannelSelect,
+                       channel_types=[discord.ChannelType.text, discord.ChannelType.private],
+                       min_values=1, max_values=1)
+    async def select_callback(self, itx: discord.Interaction[LightningBot], select: discord.ui.ChannelSelect):
+        channel = select.values[0].resolve()
+        await itx.response.defer()
+        if not channel:
+            await itx.followup.send("Unable to set the verification channel. Please try again!", ephemeral=True)
+            return
+
+        query = """INSERT INTO guild_gatekeeper_config (guild_id, verification_channel_id)
+                   VALUES ($1, $2)
+                   ON CONFLICT (guild_id)
+                   DO UPDATE SET verification_channel_id=EXCLUDED.verification_channel_id;"""
+        await itx.client.pool.execute(query, itx.guild_id, channel.id)
+        confirm = ConfirmationView("", author_id=itx.user.id)
+        msg = await itx.followup.send(f"Set the verification channel to {channel.mention}! Would you like me to "
+                                      "set up the channel permissions for you? Gatekeeper requires that everyone "
+                                      f"cannot read this channel, only the {self.role.mention} can read messages",
+                                      view=confirm, ephemeral=True, wait=True)
+        await confirm.wait()
+        if not confirm.value:
+            return
+
+        try:
+            default_p = channel.permissions_for(itx.guild.default_role)
+            if default_p.read_messages:
+                overwrites = channel.overwrites_for(itx.guild.default_role)
+                overwrites.read_messages = False
+                await channel.set_permissions(itx.guild.default_role, overwrite=overwrites)
+            if not channel.permissions_for(self.role).read_messages:
+                overwrite = channel.overwrites_for(self.role)
+                overwrite.read_messages = True
+                overwrite.read_message_history = True  # sometimes a lil' silly permissions
+                overwrite.send_messages = False
+                overwrite.add_reactions = False
+                overwrite.create_public_threads = False
+                overwrite.create_private_threads = False
+                overwrite.send_messages_in_threads = False
+                overwrite.use_application_commands = False
+                await channel.set_permissions(self.role, overwrite=overwrite)
+        except discord.HTTPException as e:
+            await msg.edit(content=f"Unable to set channel permissions ({e})", view=None)
+            self.stop(interaction=itx)
+
+        await msg.edit(content="Set the correct permissions for the verification channel!", view=None)
+        self.stop(interaction=itx)
+
+
+class GatekeeperSetup(UpdateableMenu, ExitableMenu):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Test `GatekeeperSetup` for correct initial message formatting and button functionality.

Since `GatekeeperSetup` is central to configuring the gatekeeper feature, it's important to test that it correctly formats the initial message based on the gatekeeper configuration and that all buttons function as expected.